_posts/2018-08-30-on-the-fly-transformations.md

```diff
@@ -0,0 +1,342 @@
+ fall outside the unit cell ;
+ - shift the molecules (`compound='fragments'`) back to the unit cell using `wrap`
+
+This is how it looks:
+
+```python
+u = mda.Universe('pept_in_memb.tpr', 'pept_in_memb.xtc')
+prot = u.select_atoms("protein")
+ag = u.atoms
+# we will use mass as weights for the center calculation
+transformations = (mda.transformations.unwrap(ag),
+                   mda.transformations.center_in_box(prot, weights='mass'),
+                   mda.transformations.wrap(ag, compound='fragments'))
+u.trajectory.add_transformations(*transformations)
+nv.show_mdanalysis(u)
+```
+ 
+Two other centering transformations are availabe - `center_in_plane` and `center_in_axis` - and just as
+the names say, they are used to center molecules in the `xy`, `xz` and `yz` planes, and the `x`, `y` and `z`
+axes, respectively.
+ 
+### Example 3: what if we want to do a fitting of the protein?
+Fitting is useful when processing trajectories for visualization and analyses - it removes the translations
+and rotations of the molecule, allowing us to have a better look at the structural changes that happen in
+our simulations.
+If we want to do this using `trjconv` we would do have to do this in two steps:
+
+     gmx trjconv -f pept_in_memb.xtc -s pept_in_memb.tpr -pbc mol -center -o midstep.xtc
+    
+     gmx trjconv -f midstep.xtc -s pept_in_memb.tpr -fit rot+trans -o output.xtc
+
+And we choose `Protein` as the group to be centered and for the least squares fitting.
+
+In MDAnalysis we just add another transformation to our workflow - `fit_rot_trans`. This transformation takes
+the AtomGroup to be fitted as argument, an AtomGroup to be used as reference and, by default, it behaves just
+as the option `-fit rot+trans`. If given a `plane` argument, the fitting is performed on a given plane. If
+`plane=xy` then the transformation will behave as `-fit rotxy+transxy`, but the `xz` and `yz` planes are also
+supported. Just as in `center_in_box`, a `weights` argument can be passed to the function, and it will dictate
+how much each atom of the molecule contributes to the least squares fitting. 
+Here's what the workflow looks like:
+
+
+```python
+u = mda.Universe('pept_in_memb.tpr', 'pept_in_memb.xtc')
+prot = u.select_atoms("protein")
+# we load another universe to define the reference
+# it uses the same input files, but this doesn't have to be always the case
+ref_u = mda.Universe('pept_in_memb.tpr', 'pept_in_memb.xtc')
+reference = ref_u.select_atoms("protein")
+ag = u.atoms
+transformations = (mda.transformations.unwrap(ag),
+                   mda.transformations.center_in_box(prot, weights='mass'),
+                   mda.transformations.wrap(ag, compound='fragments'),
+                   mda.transformations.fit_rot_trans(prot, reference))
+u.trajectory.add_transformations(*transformations)
+nv.show_mdanalysis(u)
+
+```
+
+
+    NGLWidget(count=11)
```

Comment:
What is this line `NGLWidget(count=11)` doing here? Is this supposed to embed the nglviewer widget??

Reply:
I think it is an artifact coming from the notebook. Removed it in the latest commit.